object-store-client-play-26/src/test/scala/uk/gov/hmrc/objectstore/client/wiremock/stubs/ObjectStoreStubs.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.objectstore.client.wiremock.stubs
+
+import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, binaryEqualTo, delete, equalTo, get, put, stubFor, urlEqualTo}
+import uk.gov.hmrc.objectstore.client.Path
+import uk.gov.hmrc.objectstore.client.play.Md5Hash
+
+object ObjectStoreStubs {
+
+  def initPutObjectStub(
+    path: Path.File,
+    statusCode: Int,
+    reqBody: Array[Byte],
+    md5Base64: String,
+    contentType: String = "application/octet-stream",
+    owner: String
+  ): Unit = {
+    val request = put(urlEqualTo(s"/object-store/object/$owner/${path.asUri}"))
+      .withHeader("Authorization", equalTo("AuthorizationToken"))
+      .withHeader("Content-Length", equalTo("49"))
+      .withHeader("Content-Type", equalTo(contentType))
+      .withHeader("Content-MD5", equalTo(md5Base64))
+      .withRequestBody(binaryEqualTo(reqBody))
+
+    val response = aResponse().withStatus(statusCode)
+    stubFor(
+      request
+        .willReturn(response))
+  }
+
+  def initListObjectsStub(
+    path: String,
```

Comment:
since the type has changed from `Path.Directory` to `String` - it is at odds with the other stubs, which use the model.
If you want to be asserting that the url is correctly constructed, then we probably shouldn't be passing in owner as a separate param either.
Why not overload? `def initListObjectStub(path: Path.Directory, owner: String) = initListObjectStub(url = ".../$owner/${path.asUri}")` and your new test when you particularly want to check that the url has been constructed properly calls with `"url = ".../$owner/"`

Reply:
The reason I changed the type from `Path.Directory` to `String` is that I don't want to affect the bugs from production code to affect the wiremock expectations. When I wrote the test for this scenario, it passed on its own because wiremock stubbing was using `asUri`. I would rather like to clearly specify what I expect rather depend on model class for expectation.

I am not in favour of using model classes for doing setup/assertions where external components (in this case - wiremock) are involved.

Reply:
Fixed